fennel/datasets/datasets.py

```diff
@@ -594,8 +599,14 @@ def dataset(
         file_name = ""
 
     def _create_lookup_function(
-        cls_name: str, key_fields: List[str]
+        cls_name: str, key_fields: List[str], struct_types: Dict[str, Any] = {}
```

Comment:
It's usually a bad idea to use mutable data (e.g. lists, dicts) as default arguments. Better to make it None and convert it to empty dict when None.